tests/unit/utils/test_config.py

```diff
@@ -359,6 +359,88 @@ def test_invalid_config():
         "llm_temperature_override must be between 0 and 1",
     )
 
+    check_expected_exception(
+        """
+---
+llm_providers:
+  - name: p1
+    credentails_path: tests/config/secret.txt
+    models:
+      - name: m1
+        credentails_path: tests/config/secret.txt
+ols_config:
+  reference_content:
+    product_docs_index_path: "./invalid_dir"
+    product_docs_index_id: product
+  conversation_cache:
+    type: memory
+    memory:
+      max_entries: 1000
+dev_config:
+  llm_temperature_override: 0.1
+  enable_dev_ui: true
+  disable_question_validation: false
+  disable_auth: false
+
+""",
+        InvalidConfigurationError,
+        "Reference content path './invalid_dir' does not exist",
+    )
+
+    check_expected_exception(
+        """
+---
+llm_providers:
+  - name: p1
+    credentails_path: tests/config/secret.txt
+    models:
+      - name: m1
+        credentails_path: tests/config/secret.txt
+ols_config:
+  reference_content:
+    product_docs_index_path: "/tmp"
+  conversation_cache:
+    type: memory
+    memory:
+      max_entries: 1000
+dev_config:
+  llm_temperature_override: 0.1
+  enable_dev_ui: true
+  disable_question_validation: false
+  disable_auth: false
+
+""",
+        InvalidConfigurationError,
+        "product_docs_index_path is specified but product_docs_index_id is missing",
```

Comment:
once you add the validation logic, you'll need the inverse test case:  index id specified but index path missing.